### PR TITLE
Adding vspc deployment file

### PR DESCRIPTION
**Pull request by rajivmucheli**
we created vmware-vspc-deployment.yaml, as there was a change in plan and opted for only 1 file. vmware-vspc-service.yaml has to be created.

### CODE REVIEW

File: nova/templates/vmware-vspc-deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+      labels:
+        name: nova-vmware-vspc
+      annotations:
+        pod.beta.kubernetes.io/hostname:  nova-vmware-vspc
+    spec:
+      containers:
+        - name: nova-vmware-vspc
+          image: {{.Values.global.image_repository}}/{{.Values.global.image_namespace}}/ubuntu-source-vspc:{{.Values.image_version_vspc}}
+          imagePullPolicy: IfNotPresent
+spec:
```
Comment (fwiesel) [inline nova/templates/vmware-vspc-deployment.yaml]: Hi,

the spec following the is a service definition.
Can you please move it to a separate file, and set the correct `kind`, `apiVersion` and `metadata` for it.

**Comment by rajivmucheli** [comment]
Updated the comments and created vmware-vspc-service.yaml

File: nova/templates/vmware-vspc-service.yaml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+metadata:
+  name: nova-vmware-vspc
+  labels:
+    system: openstack
+    type: backend
+    component: nova
+spec:
+  type: NodePort
+   selector:
+   name: nova-vmware-vspc
```
Comment (fwiesel) [inline nova/templates/vmware-vspc-service.yaml]: This has to be intended by one. It is an attribute of the selector.
The selector of the nodeport should choose all pods with the name `nova-vmware-vspc`

File: nova/templates/vmware-vspc-deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    metadata:
+      labels:
+        name: nova-vmware-vspc
+      annotations:
+        pod.beta.kubernetes.io/hostname:  nova-vmware-vspc
+    spec:
+      containers:
+        - name: nova-vmware-vspc
+          image: {{.Values.global.image_repository}}/{{.Values.global.image_namespace}}/ubuntu-source-vspc:{{.Values.image_version_vspc}}
+          imagePullPolicy: IfNotPresent
```
Comment (fwiesel) [inline nova/templates/vmware-vspc-deployment.yaml]: The container is missing a specification of the ports it is exposing.
Check here for how it works: https://kubernetes.io/docs/tutorials/connecting-apps/connecting-frontend-backend/

File: nova/templates/vmware-vspc-service.yaml [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+spec:
+   selector:
+   name: nova-vmware-vspc
+   ports:
+     - name: proxy
+       port: 13370
+       targetPort: proxy
+    - name: admin
+      port: 13371 
+       targetPort: admin
```
Comment (fwiesel) [inline nova/templates/vmware-vspc-service.yaml]: Here the indentation is too deep.

File: nova/templates/vmware-vspc-service.yaml [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+  externalIPs: ["10.44.11.151"]
```
Comment (fwiesel) [inline nova/templates/vmware-vspc-service.yaml]: This should be a variable, which we will define in the regional setup

File: vmware-vspc-deployment.yaml [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+      labels:
+        name: nova-vmware-vspc
+      annotations:
+        pod.beta.kubernetes.io/hostname:  nova-vmware-vspc
+    spec:
+      containers:
+        - name: nova-vmware-vspc
+          image: {{.Values.global.image_repository}}/{{.Values.global.image_namespace}}/ubuntu-source-vspc:{{.Values.image_version_vspc}}
+          imagePullPolicy: IfNotPresent
+spec:
```
Comment (fwiesel) [inline vmware-vspc-deployment.yaml]: The spec for the node-port has to be removed from this file

File: nova/templates/vspc-configmap.yaml [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+kind: ConfigMap
+metadata:
+  name: nova-vmware-vspc-config
+  labels:
+    system: openstack
+    type: config
+    component: nova 
+data:
+  config.json: |
+{{ tuple "etc/_ config.json.tpl.tpl" . | include "template" | indent 4 }}
```
Comment (fwiesel) [inline nova/templates/vspc-configmap.yaml]: Could you move the config.json and the vspc.conf  into this file?

**Review by fwiesel — CHANGES_REQUESTED** [review]

File: nova/templates/vspc-configmap.yaml [hunk 8]
```diff
@@ -0,0 +1,12 @@
...
+kind: ConfigMap
+metadata:
+  name: nova-vmware-vspc-config
+  labels:
+    system: openstack
+    type: config
+    component: nova 
+data:
+  config.json: |
+   command : "vmware-vspc --config-file vspc.conf"
```
Comment (fwiesel) [inline nova/templates/vspc-configmap.yaml]: Hi, now the config.json will only contain
`command : "vmware-vspc --config-file vspc.conf"`, which is not a proper json format.

**Comment by fwiesel** [comment]
Closing as the project is deprecated in favour of sapcc/helm-charts

**Pull request closed without merging**